pkg/artifactory/resource/webhook/resource_artifactory_webhook_base.go

```diff
@@ -28,7 +28,19 @@ var baseCriteriaSchema = map[string]*schema.Schema{
 	},
 }
 
-var baseWebhookBaseSchema = func(webhookType string) map[string]*schema.Schema {
+var getBaseSchemaByVersion = func(webhookType string, version int) map[string]*schema.Schema {
+	var baseSchema map[string]*schema.Schema
+
+	if version == 1 {
+		baseSchema = baseWebhookBaseSchemaV1(webhookType)
+	} else {
+		baseSchema = baseWebhookBaseSchemaV2(webhookType)
+	}
+
+	return baseSchema
+}
+
+var baseWebhookBaseSchemaV1 = func(webhookType string) map[string]*schema.Schema {
```

Comment:
since these are not in a closure scope (and, I assume that's what you want) then don't use `var` - declare a normal function